model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -14,67 +14,223 @@
+        # so reshape-able CTCGreedyDecoderReplacement2 transformation is applied first
+        return [CTCGreedyDecoderReplacement2]
+
+    @staticmethod
+    def pattern(**kwargs):
+        return dict(
+            nodes=[('decoder', dict(op='CTCGreedyDecoder')),
+                   ('cast', dict(op='Cast')),
+                   ('sparse_to_dense', dict(op='SparseToDense'))
+                   ],
+            edges=[('decoder', 'sparse_to_dense', {'out': 0}),
+                   ('decoder', 'cast', {'out': 1}),
+                   ('cast', 'sparse_to_dense', {'out': 0})
+                   ]
+        )
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        # TODO: Once Inference Engine's CTCGreedyDecoder starts to support sequence length format like in TensorFlow,
+        # CTCGreedyDecoderReplacement2 needs to be removed and CTCGreedyDecoderReplacement, a more generic
+        # transformation, needs to be adopted for all cases
+        ctc_greedy_decoder = match['decoder']
+        cast = match['cast']
+        sparse_to_dense = match['sparse_to_dense']
+        sparse_to_dense_name = sparse_to_dense.soft_get('name', sparse_to_dense.id)
+
+        # disconnect SparseToDense and Cast nodes
+        sparse_to_dense.in_port(0).disconnect()
+        cast.in_port(0).disconnect()
+
+        # transform CTCGreedyDecoder output to TensorFlow's one:
+        # 1. squeeze the output to [N, T] shape
+        # 2. cast it to integer
+        squeeze_dec_seq = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([2, 3])},
+                                                      {'name': sparse_to_dense_name})
+        squeeze_dec_seq.in_port(0).connect(ctc_greedy_decoder.out_port(0))
+        cast_to_int = Cast(graph, {'name': sparse_to_dense_name + '/CastToInt',
+                                   'dst_type': np.int32}).create_node()
+        cast_to_int.in_port(0).connect(squeeze_dec_seq.out_port(0))
+
+        # preserve output name from original graph
+        rename_nodes([(sparse_to_dense, sparse_to_dense_name + '/AbandonedName'),
+                      (cast_to_int, sparse_to_dense_name)])
+
+        # set output of the new sub-graph as a source for SparseToDense consumer
+        sparse_to_dense.out_port(0).get_connection().set_source(cast_to_int.out_port(0))
+
+        # remove no longer needed nodes
+        graph.remove_nodes_from([sparse_to_dense.id, cast.id])
+
+        # mark CTCGreedyDecoder node as a node that requires transformation of sequence length to a mask format
+        # in the middle phase
+        ctc_greedy_decoder['use_mask_format'] = True
+
+        # unless the second input of CTCGreedyDecoder is a parameter, it enforces MO to use --static-shape
+        # to try getting the second input with a value
+        sequence_length_node = ctc_greedy_decoder.in_node(1)
+        if sequence_length_node.op != 'Parameter' and not graph.graph['cmd_params'].static_shape:
```

Comment:
Please, change `sequence_length_node.op` to `sequence_length_node.soft_get('op')`. It is a bit more reliable.

Reply:
Fixed.